benchmarks/helpers/main.js

```diff
@@ -14,17 +24,23 @@ export const timeFunction = async (func, runs = 1) => {
 
   observer.observe({ entryTypes: ['measure'] })
 
-  await Promise.all(
-    Array.from({ length: runs }).map(async (_, index) => {
-      performance.mark(`run-${index}-start`)
+  for (let index = 0; index < runs; index++) {
+    performance.mark(`run-${index}-start`)
+
+    await func(index)
 
-      await func(index)
+    performance.measure(`run-${index}`, `run-${index}-start`)
+  }
 
-      performance.measure(`run-${index}`, `run-${index}-start`)
-    }),
-  )
+  // wait for PerformanceObserver to gather all data
+  await sleep(100)
```

Comment:
As the PerformanceObserver is triggered async, we need to wait till it was triggered.
Right now out of the 3 runs we usually only calculate the average with the numbers of 2 runs but divide it by 3.

The assert a little down is also there to ensure we captured all runs.